ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -351,12 +351,15 @@ private static PhysicalSchema getPhysicalSchema(final SourceStep<?> streamSource
         // re-partitioned topic will be used for any subsequent state stores, in lieu
         // of the original source topic, thus avoiding the issues above.
         // See https://github.com/confluentinc/ksql/issues/6650
-        table = source.mapValues(row -> row);
+        table = source.mapValues(row -> row, Materialized.with(null, valueSerde));
```

Comment:
Ensure that we don't drop the value serde (similar below)

Reply:
I'm confused -- is this creating a new state store? We need to ensure that any changes here are backwards compatible with existing plans.

Reply:
Hmm... Good point. Need to think about it.

Reply:
Before we fix https://issues.apache.org/jira/browse/KAFKA-12242, maybe we should just pass `valueSerde` along to downstream and then set when needed later.

Reply:
It should actually be fine -- as long as we don't set a store name, we won't force materialization. Will revert this anyway, and just reconstruct the Serde in the FK-join node instead.

Reply:
Sounds good.